backend/ASTU-backend-group-1/delivery/controllers/user_controllers.go

```diff
@@ -0,0 +1,155 @@
+package controllers
+
+import (
+	"astu-backend-g1/domain"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+type UserController struct {
+	userUsecase domain.UserUsecase
+}
+
+func NewUserController(userUsecase domain.UserUsecase) *UserController {
+	return &UserController{userUsecase: userUsecase}
+}
+
+func (c *UserController) Register(ctx *gin.Context) {
+	user := domain.User{}
+	if err := ctx.ShouldBind(&user); err != nil {
+		ctx.JSON(http.StatusNotAcceptable, gin.H{"error": err.Error()})
+		return
+	}
+	_, err := c.userUsecase.Create(&user)
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	ctx.IndentedJSON(http.StatusCreated, gin.H{"message":"Activate your Account in the your email link" })
+}	
+func (c *UserController) AccountVerification(ctx *gin.Context) {
+	email := ctx.Query("email")
+	token := ctx.Query("pwd")
+	_, err := c.userUsecase.AccountVerification(email, token)
+	if err != nil {
+		ctx.JSON(http.StatusNotAcceptable, gin.H{"error": err.Error()})
+		return
+	}
+	ctx.IndentedJSON(http.StatusOK, gin.H{"message":"Account Activated"})
+}
+func (c *UserController) ForgetPassword(ctx *gin.Context) {
+	email := ctx.Query("email")
+	_, err := c.userUsecase.ForgetPassword(email)
+	if err != nil {
+		ctx.JSON(http.StatusNotAcceptable, gin.H{"error": err.Error()})
+		return
+	}
+	ctx.IndentedJSON(http.StatusOK, gin.H{"message":"Password reset token sent to your email"})
+}
+func (c *UserController) ResetPassword(ctx *gin.Context) {
+	email := ctx.Query("email")
+	token := ctx.Query("pwd")
```

Comment:
It is better if you used the same name here for the pwd part. make both of the same it makes a lot of sense after that